x-pack/plugin/monitoring/src/internalClusterTest/java/org/elasticsearch/xpack/monitoring/integration/MonitoringIT.java

```diff
@@ -145,39 +146,45 @@ public void testMonitoringBulk() throws Exception {
                 ensureGreen(monitoringIndex);
                 assertThat(client().admin().indices().prepareRefresh(monitoringIndex).get().getStatus(), is(RestStatus.OK));
 
-                final SearchResponse response = client().prepareSearch(".monitoring-" + system.getSystem() + "-" + TEMPLATE_VERSION + "-*")
-                    .get();
+                assertResponse(client().prepareSearch(".monitoring-" + system.getSystem() + "-" + TEMPLATE_VERSION + "-*"), response -> {
+                    // exactly 3 results are expected
+                    assertThat("No monitoring documents yet", response.getHits().getTotalHits().value, equalTo(3L));
 
-                // exactly 3 results are expected
-                assertThat("No monitoring documents yet", response.getHits().getTotalHits().value, equalTo(3L));
+                    final List<Map<String, Object>> sources = Arrays.stream(response.getHits().getHits())
+                        .map(SearchHit::getSourceAsMap)
+                        .collect(Collectors.toList());
 
-                final List<Map<String, Object>> sources = Arrays.stream(response.getHits().getHits())
-                    .map(SearchHit::getSourceAsMap)
-                    .collect(Collectors.toList());
-
-                // find distinct _source.timestamp fields
-                assertThat(sources.stream().map(source -> source.get("timestamp")).distinct().count(), is(1L));
-                // find distinct _source.source_node fields (which is a map)
-                assertThat(sources.stream().map(source -> source.get("source_node")).distinct().count(), is(1L));
+                    // find distinct _source.timestamp fields
+                    assertThat(sources.stream().map(source -> source.get("timestamp")).distinct().count(), is(1L));
+                    // find distinct _source.source_node fields (which is a map)
+                    assertThat(sources.stream().map(source -> source.get("source_node")).distinct().count(), is(1L));
+                });
             });
 
             final SearchResponse response = client().prepareSearch(monitoringIndex).get();
-            final SearchHits hits = response.getHits();
+            try {
```

Comment:
Why not use the `assertResponse` pattern here?

Reply:
I tried, but when I did that, the `assertMonitoringDoc` method (currently line 184) could no longer be resolved for reasons that weren't clear to me. And (more minor) `toMap` said it threw an unhandled Exception, so it would have been clunky to handle as well.

Reply:
Fair point lets leave it this way :)

Reply:
You can use `assertCheckedResponse` and then you don't need to manually do it in the test.

Reply:
Excellent. Thanks for the tip.